scripts/create_github_releases/main.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	// tag must specify major.minor.patch or major.minor.patch-rc#
+	// (e.g. 1.12.1, 1.12.1-rc1)
+	validTag = regexp.MustCompile(`^\d+\.\d+\.\d+(-rc\d+)?$`)
+	// target must specify release branch for releasing
+	// or main for release candidates
+	validTarget = regexp.MustCompile(`^(release-\d+-\d+|main)$`)
+	// true or false
+	validPrerelease = regexp.MustCompile(`^(true|false)$`)
+	// yes or no
+	validYesNo = regexp.MustCompile(`^(yes|no)$`)
```

Comment:
Suggest you const the expressions, and compile the regexps where you use them, rather than declare globals.